develop/2016-12-15-thiagobrito-nepotism-spends.py

```diff
@@ -0,0 +1,358 @@
+
+# coding: utf-8
+
+# # Find clusters of politicians spending with companies owned by each others relatives or himself (issue #18)
+
+# In this analysis we try to identify the pattern described bellow:
+# 
+# * According to "ATO DA MESA Nº 43, DE 21/5/2009 Art. 4º § 13", no expense can be reimbursed if made with companies where the applicant (politician) is owner, partner or relative (maximum of third degree).
+# 
+# Politicians may know about this and have groups to "exchange expenses" to cheat the law. e.g. Politician A pays for Politician B's relatives; Politician B pays for Politician A's relatives.
+
+# In[1]:
+
+from unicodedata import normalize
+def normalize_name(name):
+    return normalize('NFKD', name).encode('ASCII', 'ignore').decode('utf-8').upper()
+
+def normalize_list(my_list):
+    result = []
+    for n in my_list:
+        result.append(normalize_name(n))
+    return result
+
+import requests
+import json
+import os
+import time
+
+cache = {}
+cache_path = '../data/cnpj_cache.json'
+if os.path.exists(cache_path):
+    cache = json.loads(open(cache_path, 'r').read())
+
+def save_cnpj_cache():
+    global cache
+    open(cache_path, 'w').write(json.dumps(cache))
+    
+def extract_cnpj_board(cnpj):
+    global cache
+    result = []
+    try:
+        if cnpj not in cache:
+            url = 'http://receitaws.com.br/v1/cnpj/%s' % cnpj
```

Comment:
You can also use a more recent one that doesn't have geolocation: `2017-05-21-companies-no-geolocation.xz`